modules/metrics/src/main/scala/com/evolutiongaming/skafka/metrics/KafkaMetricsCollector.scala

```diff
@@ -61,39 +64,48 @@ class KafkaMetricsCollector[F[_]: Monad: ToTry](kafkaClientMetrics: F[Seq[Client
 
   override def collect(): util.List[MetricFamilySamples] = {
     for {
-      metrics <- kafkaClientMetrics
+      metrics      <- kafkaClientMetrics
       metricsGroups = metrics.groupBy(m => (m.name, m.group)).values.toList
-      result <- metricsGroups.traverse { metricsGroup =>
-        val metric = metricsGroup.head
-        val prometheusName = getPrometheusName(metric)
-        val collectorType = getCollectorType(metric)
+      result <- metricsGroups
+        .traverse { metricsGroup =>
+          val metric         = metricsGroup.head
+          val prometheusName = getPrometheusName(metric)
+          val collectorType  = getCollectorType(metric)
 
-        prometheusName match {
-          case Some(name) =>
-            metricsGroup.toVector.traverse { metric =>
-              val tags = metric.tags.flatMap {
-                case (key, value) =>
-                  val prometheusKey = key.replaceAll("-", "_")
-                  if (LabelNameRegex.findFirstIn(prometheusKey).contains(prometheusKey)) (prometheusKey -> value).some else None
-              }
-              val tagsKeys = tags.keys.toList.asJava
-              val tagsValues = tags.values.toList.asJava
-              metric.value.map {
-                case v: Number =>
-                  new Sample(name, tagsKeys, tagsValues, v.doubleValue()).some
+          prometheusName match {
+            case Some(name) =>
+              metricsGroup
+                .toVector
+                .traverse { metric =>
+                  val tags = metric.tags.flatMap {
+                    case (key, value) =>
+                      val prometheusKey = key.replaceAll("-", "_")
+                      if (LabelNameRegex.findFirstIn(prometheusKey).contains(prometheusKey))
+                        (prometheusKey -> value).some
+                      else None
+                  }
+                  val (customLabelsKeys, customLabelsValues) = customLabels.separate
```

Comment:
seems like it doesn't have to be called on each `collect`

Reply:
Good catch, changed